holidays/countries/thailand.py

```diff
@@ -10,24 +10,138 @@
+        1999: ((DEC, 6, thai_special_in_lieu_holidays_en),),
+        2000: ((JAN, 3, thai_special_in_lieu_holidays_en),),
+        2000: ((FEB, 21, thai_special_in_lieu_holidays_en),),
+        2000: ((AUG, 14, thai_special_in_lieu_holidays_en),),
+        2000: ((DEC, 11, thai_special_in_lieu_holidays_en),),
+        ### From 2001 Onwards (Checked /w Bank of Thailand Data) ###
+        2006: ((APR, 19, thai_election_en),),
+        2006: ((JUN, 9, rama_nine_sixty_accession_en),),
+        2006: ((JUN, 12, rama_nine_sixty_accession_en),),
+        2006: ((JUN, 13, rama_nine_sixty_accession_en),),
+        2006: ((SEP, 20, thai_military_emergency_lockdown_en),),
+        2007: ((DEC, 24, thai_election_in_lieu_en),),
+        2009: ((JAN, 2, thai_bridge_public_holiday_en),),
+        2009: ((APR, 10, thai_political_emergency_lockdown_en),),
+        2009: ((APR, 16, thai_political_emergency_lockdown_en),),
+        2009: ((APR, 17, thai_political_emergency_lockdown_en),),
+        2009: ((JUL, 6, thai_bridge_public_holiday_en),),
+        2010: ((MAY, 20, thai_bridge_public_holiday_en),),
+        2010: ((MAY, 21, thai_bridge_public_holiday_en),),
+        2010: ((AUG, 13, thai_bridge_public_holiday_en),),
+        2011: ((MAY, 16, thai_bridge_public_holiday_en),),
+        2011: ((OCT, 27, thai_flood_2011_emergency_lockdown_en),),
+        2011: ((OCT, 28, thai_flood_2011_emergency_lockdown_en),),
+        2011: ((OCT, 31, thai_flood_2011_emergency_lockdown_en),),
+        2012: ((JAN, 3, thai_bridge_public_holiday_en),),
+        2012: ((APR, 9, thai_bridge_public_holiday_en),),
+        2013: ((DEC, 30, thai_bridge_public_holiday_en),),
+        2014: ((AUG, 11, thai_bridge_public_holiday_en),),
+        2015: ((JAN, 2, thai_bridge_public_holiday_en),),
+        2015: ((MAY, 4, thai_bridge_public_holiday_en),),
+        2016: ((MAY, 6, thai_bridge_public_holiday_en),),
+        2016: ((JUL, 18, thai_bridge_public_holiday_en),),
+        2016: ((OCT, 14, rama_nine_mourning_en),),
+        2017: ((OCT, 26, rama_nine_cremation_en),),
+        2019: ((MAY, 6, rama_ten_coronation_celebrations_en),),
+        2020: ((JUL, 27, songkran_festival_in_lieu_covid_en),),
+        2020: ((SEP, 4, songkran_festival_in_lieu_covid_en),),
+        2020: ((SEP, 7, songkran_festival_in_lieu_covid_en),),
+        2020: ((NOV, 19, thai_bridge_public_holiday_en),),
+        2020: ((NOV, 20, thai_bridge_public_holiday_en),),
+        2020: ((DEC, 11, thai_bridge_public_holiday_en),),
+        2021: ((FEB, 12, thai_bridge_public_holiday_en),),
+        2021: ((APR, 12, thai_bridge_public_holiday_en),),
+        2021: ((SEP, 24, thai_bridge_public_holiday_en),),
+        2022: ((JUL, 15, thai_bridge_public_holiday_en),),
+        2022: ((JUL, 29, thai_bridge_public_holiday_en),),
+        2022: ((OCT, 14, thai_bridge_public_holiday_en),),
+        2022: ((DEC, 30, thai_bridge_public_holiday_en),),
+        2023: ((MAY, 5, thai_bridge_public_holiday_en),),
+    }
```

Comment:
```suggestion
    special_holidays = {
        ### 1992-1994 (include In Lieus, Checked /w Bank of Thailand Data) ###
        1992: (
            (MAY, 18, thai_special_in_lieu_holidays_en),
            (DEC, 7, thai_special_in_lieu_holidays_en),
        ),
        1993: (
            (MAR, 8, thai_special_in_lieu_holidays_en),
            (MAY, 3, thai_special_in_lieu_holidays_en),
            (OCT, 25, thai_special_in_lieu_holidays_en),
            (DEC, 6, thai_special_in_lieu_holidays_en),
        ),
        1994: (
            (JAN, 3, thai_special_in_lieu_holidays_en),
            (MAY, 2, thai_special_in_lieu_holidays_en),
            (JUL, 25, thai_special_in_lieu_holidays_en),
            (OCT, 24, thai_special_in_lieu_holidays_en),
            (DEC, 12, thai_special_in_lieu_holidays_en),
        ),
        ### 1995-1997 (Bank of Thailand Data) ###
        1996: ((JUN, 10, rama_nine_golden_jubilee_en),),
        ### 1998-2000 (include In Lieus, Checked /w Bank of Thailand Data) ###
        1998: (
            (MAY, 11, thai_special_in_lieu_holidays_en),
            (DEC, 7, thai_special_in_lieu_holidays_en),
        ),
        1999: (
            (MAY, 3, thai_special_in_lieu_holidays_en),
            (MAY, 31, thai_special_in_lieu_holidays_en),
            (OCT, 25, thai_special_in_lieu_holidays_en),
            (DEC, 6, thai_special_in_lieu_holidays_en),
        ),
        2000: (
            (JAN, 3, thai_special_in_lieu_holidays_en),
            (FEB, 21, thai_special_in_lieu_holidays_en),
            (AUG, 14, thai_special_in_lieu_holidays_en),
            (DEC, 11, thai_special_in_lieu_holidays_en),
        ),
        ### From 2001 Onwards (Checked /w Bank of Thailand Data) ###
        2006: (
            (APR, 19, thai_election_en),
            (JUN, 9, rama_nine_sixty_accession_en),
            (JUN, 12, rama_nine_sixty_accession_en),
            (JUN, 13, rama_nine_sixty_accession_en),
            (SEP, 20, thai_military_emergency_lockdown_en),
        ),
        2007: ((DEC, 24, thai_election_in_lieu_en),),
        2009: (
            (JAN, 2, thai_bridge_public_holiday_en),
            (APR, 10, thai_political_emergency_lockdown_en),
            (APR, 16, thai_political_emergency_lockdown_en),
            (APR, 17, thai_political_emergency_lockdown_en),
            (JUL, 6, thai_bridge_public_holiday_en),
        ),
        2010: (
            (MAY, 20, thai_bridge_public_holiday_en),
            (MAY, 21, thai_bridge_public_holiday_en),
            (AUG, 13, thai_bridge_public_holiday_en),
        ),
        2011: (
            (MAY, 16, thai_bridge_public_holiday_en),
            (OCT, 27, thai_flood_2011_emergency_lockdown_en),
            (OCT, 28, thai_flood_2011_emergency_lockdown_en),
            (OCT, 31, thai_flood_2011_emergency_lockdown_en),
        ),
        2012: (
            (JAN, 3, thai_bridge_public_holiday_en),
            (APR, 9, thai_bridge_public_holiday_en),
        ),
        2013: ((DEC, 30, thai_bridge_public_holiday_en),),
        2014: ((AUG, 11, thai_bridge_public_holiday_en),),
        2015: (
            (JAN, 2, thai_bridge_public_holiday_en),
            (MAY, 4, thai_bridge_public_holiday_en),
        ),
        2016: (
            (MAY, 6, thai_bridge_public_holiday_en),
            (JUL, 18, thai_bridge_public_holiday_en),
            (OCT, 14, rama_nine_mourning_en),
        ),
        2017: ((OCT, 26, rama_nine_cremation_en),),
        2019: ((MAY, 6, rama_ten_coronation_celebrations_en),),
        2020: (
            (JUL, 27, songkran_festival_in_lieu_covid_en),
            (SEP, 4, songkran_festival_in_lieu_covid_en),
            (SEP, 7, songkran_festival_in_lieu_covid_en),
            (NOV, 19, thai_bridge_public_holiday_en),
            (NOV, 20, thai_bridge_public_holiday_en),
            (DEC, 11, thai_bridge_public_holiday_en),
        ),
        2021: (
            (FEB, 12, thai_bridge_public_holiday_en),
            (APR, 12, thai_bridge_public_holiday_en),
            (SEP, 24, thai_bridge_public_holiday_en),
        ),
        2022: (
            (JUL, 15, thai_bridge_public_holiday_en),
            (JUL, 29, thai_bridge_public_holiday_en),
            (OCT, 14, thai_bridge_public_holiday_en),
            (DEC, 30, thai_bridge_public_holiday_en),
        ),
        2023: ((MAY, 5, thai_bridge_public_holiday_en),),
    }
```

Reply:
Thanks, this looks way better than my current one and is also clearer at a glance